intra/protect/xdial.go

```diff
@@ -149,6 +156,8 @@ func (d *RDial) DialTCP(network string, laddr, raddr *net.TCPAddr) (*net.TCPConn
 		return nil, err
 	} else if tc, ok := c.(*net.TCPConn); ok {
 		// d.Dialer.LocalAddr = nil
+		//adjust keepalive config to save battery
+		setKeepAliveConfig(tc, defaultKeepIdle, defaultKeepCnt, defaultKeepIntvl)
```

Comment:
Another way to do so would be to create a `intra/protect/protect.go:MakeNsDialerExt` (similar to `intra/protect/protect.go:MakeNsListenConfigExt`) to pass `Control` fns to the underlying dialer from `ipn/Base.go` and `ipn/Exit.go` (that is, the sock opts should be set by the proxy dialers and not commonly by this "adapter" `xdial.go`).

https://github.com/celzero/firestack/blob/7ea0539e57ba513065dcde893084f70cfedd74b9/intra/protect/protect.go#L176

https://github.com/celzero/firestack/blob/7ea0539e57ba513065dcde893084f70cfedd74b9/intra/ipn/base.go#L42

https://github.com/celzero/firestack/blob/7ea0539e57ba513065dcde893084f70cfedd74b9/intra/ipn/exit.go#L31

Reply:
🙃Go still overrides these options.